Assets/MRTK/SDK/Experimental/Elastic/Scripts/ElasticTypes.cs

```diff
@@ -42,7 +39,77 @@ public struct ElasticExtentProperties<T>
         /// Points inside the extent to which the system will snap.
         /// </value>
         [SerializeField]
-        public T[] SnapPoints;
+        public float[] SnapPoints;
+
+        /// <value>
+        /// Distance at which snap points begin forcing the spring.
+        /// </value>
+        [SerializeField]
+        public float SnapRadius;
+    }
+
+    /// <summary>
+    /// Properties of a three-dimensional extent
+    /// in which a damped harmonic oscillator is free to move.
+    /// </summary>
+    [Serializable]
+    public struct VolumeElasticExtent
+    {
+        /// <value>
+        /// Represents the lower bound of the extent.
+        /// </value>
+        [SerializeField]
+        public Bounds StretchBounds;
+
+        /// <summary>
+        /// Whether the bounds should be respected by the system.
+        /// </summary>
+        [SerializeField]
+        public bool UseBounds;
+
+        /// <value>
+        /// Points inside the extent to which the system will snap.
+        /// </value>
+        [SerializeField]
+        public Vector3[] SnapPoints;
+
+        /// <value>
+        /// Should the SnapPoints be repeated to infinity to serve
+        /// as regular snapping intervals?
+        /// </value>
+        public bool repeatSnapPoints;
```

Comment:
Can you add some more details to the comments to explain how the snap points will be repeated to infinity? i.e. if you have points (0,0,0), (1, 1, 1) (0, 1, 0) are they all repeated somehow by taking the convex hull of all of those points and trying to stack them next to each other? Or are only a certain set of points stacked? Does snap radius come into play (i.e. the way they are repeated to infinity?)

Reply:
Ah, yes, I can clarify that.